src/web/components/elements/InteractiveBlockComponent.tsx

```diff
@@ -57,18 +63,171 @@ and is sent with all interactive elements in the scriptUrl from CAPI to do a two
+
+const placeholderLinkStyle = css`
+	position: absolute;
+	bottom: 1rem;
+	left: 1rem;
+`;
+
+// https://interactive.guim.co.uk/embed/iframe-wrapper/0.1/boot.js
+const setupWindowListeners = (iframe: HTMLIFrameElement) => {
+	// Calls func on trailing edge of the wait period
+
+	const postMessage = (message: Record<string, unknown>) => {
+		if (iframe.contentWindow) {
+			iframe.contentWindow.postMessage(JSON.stringify(message), '*');
+		}
+	};
 
-export const InteractiveBlockComponent = ({ url, scriptUrl, alt }: Props) => {
+	window.addEventListener(
+		'message',
+		(event) => {
+			if (event.source !== iframe.contentWindow) {
+				return;
+			}
+
+			const message: Record<string, unknown> = JSON.parse(event.data);
+
+			const postPositionMessage = (subscribe?: boolean) => {
+				const iframeBox = iframe.getBoundingClientRect();
+				postMessage({
+					id: message.id || '',
+					type: message.type || '',
+					subscribe: !!subscribe,
+					iframeTop: iframeBox.top,
+					iframeRight: iframeBox.right,
+					iframeBottom: iframeBox.bottom,
+					iframeLeft: iframeBox.left,
+					innerHeight: window.innerHeight,
+					innerWidth: window.innerWidth,
+					pageYOffset: window.pageYOffset,
+				});
+			};
+
+			// Actions
+			switch (message.type) {
+				case 'set-height':
+					if (typeof message.value === 'number') {
+						iframe.height = message.value.toString();
+					}
+					break;
+				case 'navigate':
+					if (typeof message.value === 'string') {
+						document.location.href = message.value;
+					}
+					break;
+				case 'scroll-to':
+					if (
+						typeof message.x === 'number' &&
+						typeof message.y === 'number'
+					) {
+						window.scrollTo(message.x, message.y);
+					}
+					break;
+				case 'get-location':
+					postMessage({
+						id: message.id,
+						type: message.type,
+						hash: window.location.hash,
+						host: window.location.host,
+						hostname: window.location.hostname,
+						href: window.location.href,
+						origin: window.location.origin,
+						pathname: window.location.pathname,
+						port: window.location.port,
+						protocol: window.location.protocol,
+						search: window.location.search,
+					});
+					break;
+				case 'get-position':
+					postPositionMessage();
+					break;
+				case 'monitor-position':
+					// Send initial position
+					postPositionMessage(true);
+
+					// Send updated position on scroll or resize
+					window.addEventListener(
+						'scroll',
+						libDebounce(() => {
+							postPositionMessage(true);
+						}, 50),
+					);
+					window.addEventListener(
+						'resize',
+						libDebounce(() => {
+							postPositionMessage(true);
+						}, 50),
+					);
+					break;
+			}
+		},
+		false,
+	);
+};
+
+export const InteractiveBlockComponent = ({
+	url,
+	scriptUrl,
+	alt,
+	role = 'inline',
+}: Props) => {
 	const wrapperRef = useRef<HTMLDivElement>(null);
+	const placeholderLinkRef = useRef<HTMLAnchorElement>(null);
+	const [loaded, setLoaded] = useState(false);
```

Comment:
Can we use `useOnce` here for this?

Reply:
I don't think so because the state we would need to wait on is set within the `useEffect`?

Reply:
`useOnce` uses `useEffect` so the effect is the same (if that makes sense?)

Reply:
But you're using `loaded` to branch rendering too so...

Reply:
Ah yes, no that works 👍